index.js

```diff
@@ -26,6 +26,18 @@ module.exports = function () {
 		return str;
 	}
 
+	var isLastCharLower = false;
+	for (var i = 0; i < str.length; i++) {
+		var c = str.charAt(i);
+		if (isLastCharLower && c.toUpperCase() === c) {
+			str = str.substr(0, i) + '-' + str.substr(i);
+			isLastCharLower = false;
+			i++;
+		} else {
+			isLastCharLower = c.toLowerCase() === c;
+		}
+	}
```

Comment:
Can you put this in a top-level function with a good name and use it here? Very hard when just reading the code what the intent of this code bit is.